recipes/lora_finetune_single_device.py

```diff
@@ -256,6 +258,9 @@ def _setup_model(
 
         self._lora_rank = cfg_model.lora_rank
         self._lora_alpha = cfg_model.lora_alpha
+        self._lora_attn_modules = list(cfg_model.lora_attn_modules)
+        self._apply_lora_to_mlp = cfg_model.apply_lora_to_mlp
+        self._apply_lora_to_output = getattr(cfg_model, "apply_lora_to_output", False)
```

Comment:
not related to this PR, but maybe at some point we should consider replacing the ```apply_lora_to_*``` flags with just adding ```mlp``` and ```output``` to the ```lora_modules```?

Reply:
Yeah agreed, I think this is likely where we'll head eventually. One thing is that we will probably want to make LoRA in MLP more configurable (i.e. use `w1`, `w2`, `w3` (or hopefully more descriptive names) instead of `mlp`). Otherwise the relationship between e.g. `q_proj` (nn.Linear) and `mlp` (FeedForward) being in the same config is a bit confusing. Anyways this shouldn't be a huge effort to change

Reply:
I agree that a single list is more intuitive, since, AFAICT, this is just consolidated into a single list under the hood.

> or hopefully more descriptive names

Changing names later on can invalidate the saved checkpoints, so would require some versioning for backwards compatibility.